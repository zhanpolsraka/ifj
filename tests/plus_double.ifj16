class Main{
	static double a = 5;
	static void run(){
		double b = 1;
		double as = Main.a +b+3;
	}
}
