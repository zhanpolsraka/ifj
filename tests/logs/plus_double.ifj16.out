---------------FILENAME: tests/plus_double.ifj16---------------


Instruction stack item [13] -> class with name "Main"
Instruction stack item [12] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [11] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [10] -> function with name "run"
Instruction stack item [9] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [8] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [7] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 1 -> PLUS
Instruction stack item [4] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [3] -> instruction with operation 1 -> PLUS
Instruction stack item [2] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class


---------------MAKE RELATIONS--------------
                     *
                     *
                     *
------------------EXECUTE------------------

------------------OUTPUT-------------------

STEP 1 - non-func instructions
Found Main
STEP 1 - non-func instructions
INSERT 5
---------------------------------------------------
N0: type is 22, values is 5
---------------------------------------------------
STEP 1 - non-func instructions
ASSIGN 5 to 0x1cee400
---------------------------------------------------
N0: type is 22, values is 5
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
Found run()
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
Got out of run()
STEP 1 - non-func instructions
Got out of Main
STEP 2 - validating existance of run() in Main
STEP 3 - executing run() instructions
INSERT 1
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
ASSIGN 1 to 0x1cee520
---------------------------------------------------
N0: type is 22, values is 1
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSERT 5
---------------------------------------------------
N0: type is 23, values is 5
---------------------------------------------------
INSERT 1
---------------------------------------------------
N1: type is 23, values is 1
N0: type is 23, values is 5
---------------------------------------------------
PLUS
---------------------------------------------------
N0: type is 23, values is 6
---------------------------------------------------
INSERT 3
---------------------------------------------------
N1: type is 22, values is 3
N0: type is 23, values is 6
---------------------------------------------------
PLUS
---------------------------------------------------
N0: type is 23, values is 9
---------------------------------------------------
ASSIGN 9.000000 to 0x1cee860
---------------------------------------------------
N0: type is 23, values is 9
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------

-----------------END OUTPUT----------------


----------------ALL RIGHT!!!---------------

----------------SYMBOL TREE----------------

Class Main
	Variable -> a, type DOUBLE, value [5.000000]
	Function -> run
Lokal variables: 
	Variable -> b, type DOUBLE, value [1.000000]
	Variable -> a, type DOUBLE, value [9.000000]


--------------INSTRUCTION LIST-------------

Instruction stack item [13] -> class with name "Main"
Instruction stack item [12] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [11] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [10] -> function with name "run"
Instruction stack item [9] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [8] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [7] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 1 -> PLUS
Instruction stack item [4] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [3] -> instruction with operation 1 -> PLUS
Instruction stack item [2] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class



